test/recipes/30-test_evp_data/evpmac.txt

```diff
@@ -159,6 +159,7 @@ Ctrl = digestsize:13
 Key = 000102030405060708090A0B0C0D0E0F
 Input = 000102030405060708090A0B0C0D0E0F101112131415161718191A1B1C1D1E1F202122232425262728292A2B2C2D2E2F303132333435363738393A3B3C3D3E
 Output = 5150d1772f50834a503e069a973fbd7c
+Result = MAC_CTRL_ERROR
```

Comment:
The digest size error is likely ignored? So, this defaults to the 16-byte hash?

Reply:
Actually, the issue is different, and probably a bad reuse of `pkey_test_ctrl`.  What seems to happen when that function gets an error from `EVP_PKEY_CTX_ctrl_str`, it checks to see if the *value* (i.e. "13" in this case) is a known object, and decides to ignore the error if that object denotes an algorithm to be skipped...  which seems to be what happens (I have *no* idea why, haven't dived to look).

The EVP_MAC variant of the test does *NOT* ignore any control error, and that's how this got hit.

Reply:
I spoke too fast.  The EVP_MAC variant ignores any `-2` returns, as that only says it has used an unimplemented control, which is fine to ignore here.  If ignoring it is faulty, it will show through the results that will not match the criteria.

Reply:
I dug a little more, and only noticed now that while `pkey_test_ctrl` does set an error string, it really ignores *all* errors (`rv` is set to 1 everywhere, except for memory allocation issues), so that answers the question why we haven't seen this error before, as I haven't followed that pattern.  Should I?

Reply:
Ah, found it!  Error checking in `pkey_test_ctrl` was effectively incapacitated in cce65266299e2e89303a90c131e8171225a1bf88

Reply:
Urgh...  never mind, that wasn't quite it either.  The error indication pattern in `test/evp_test.c` utterly confuses me, is all...

In the end, it seems that `mac_test_run_pkey` (the original `mac_test_run`) doesn't properly check if `pkey_test_ctrl` has set an error string, and therefore ignores any error, and resets the error string to `NULL` at the end.  ...  and this turns out to be my fault, see 7e6a3025493bae3018e8a9c71b93350859b1c484 :-/

Reply:
See #7500.  This long discussion ended up in *something* valuable